### ISSUE TITLE
Using both no_mangle and export_name attributes should cause a warning

### ISSUE THREAD

**Issue by gamozolabs** [rust-lang/rust#47446]
In a simple example like:

```
#[no_mangle]
#[export_name = "asdf"]
pub extern fn foobar() {}
```

The actual exported name will be `asdf`. It seems `export_name` wins over `no_mangle` however there is no warning letting you know this occurred. It seems there is no valid use for having both of these so it's probably worth warning about.

-B

**Comment by mversic** [comment]
> It seems there is no valid use for having both of these so it's probably worth warning about.

the use is that you get function name alias. You can refer to the function inside the crate as `foobar` but dynamically link to the same function under the name `asdf` from an outside crate. The reason could be because you already have function named `asdf` defined in your crate and you want to avoid conflicts like this:

```rs
#[no_mangle]
#[export_name = "asdf"]
pub extern fn foobar() {}    // This function cannot be named `asdf` but it you want it exported as `asdf`

pub fn asdf() {
    foobar()
}
```

I admit it's kinda obscure, but it's a feature. I use this to avoid symbol conflicts of exported functions when dynamically linking multiple crates that have functions with the same name
